pkg/utils/ocm.go

```diff
@@ -213,34 +213,43 @@ func loadOCMConfig() (*Config, error) {
 	return cfg, nil
 }
 
-func getOcmConfiguration() (*Config, error) {
+func getOcmConfiguration(ocmConfigLoader func() (*Config, error)) (*Config, error) {
 	tokenEnv := os.Getenv("OCM_TOKEN")
 	urlEnv := os.Getenv("OCM_URL")
 	refreshTokenEnv := os.Getenv("OCM_REFRESH_TOKEN") // Unlikely to be set, but check anyway
 
 	config := &Config{}
-	var err error
+	var (
+		defaultConfig          *Config
+		defaultConfigLoadError error
+	)
 
 	// If neither of the token ENVS 'OCM_TOKEN', 'OCM_REFRESH_TOKEN' are set, or 'OCM_URL' is not set,
 	// use the configuration file as base.
 	// We don't want to always load this, because the user might only use environment variables.
-	if (tokenEnv == "" && refreshTokenEnv == "") || urlEnv == "" {
-		config, err = loadOCMConfig()
-		if err != nil {
-			return &Config{}, fmt.Errorf("Could not load OCM configuration file")
+	if tokenEnv == "" || refreshTokenEnv == "" || urlEnv == "" {
+		defaultConfig, defaultConfigLoadError = ocmConfigLoader()
+		if defaultConfigLoadError != nil {
+			return &Config{}, fmt.Errorf("could not load OCM configuration file")
 		}
 	}
 
 	// Overwrite with set environment variables, to allow users to overwrite
 	// their configuration file's variables
 	if tokenEnv != "" {
 		config.AccessToken = tokenEnv
+	} else {
+		config.AccessToken = defaultConfig.AccessToken
 	}
 	if urlEnv != "" {
 		config.URL = urlEnv
+	} else {
+		config.URL = defaultConfig.URL
 	}
 	if refreshTokenEnv != "" {
 		config.RefreshToken = refreshTokenEnv
+	} else {
+		config.RefreshToken = defaultConfig.RefreshToken
```

Comment:
What's the benefit here of writing to a new config object instead of overwriting the same potentially already loaded one with the environment variables (if they exist)? It seems less readable to me.

Reply:
I agree with you that it is a bit annoying. I couldn't immediately think of a way to handle optionally loading the config from the file without changing the contract of `loadOcmConfig` to only fill in unset values. But I see now that `loadOcmConfig` is only used once so I could pass in a reference to the config object. 

I also now see that `loadOcmConfig` does not throw an error if the config file is not present. I'm not sure how I feel about that, but we could use that to our advantage and invert the logic here. Is that what you were thinking? 

I think I'm about even on all three options--leaving it as is in this PR, changing to pass a `*Config` reference to `loadOcmConfig`, and overwriting the result of `loadOcmConfig` if the env vars are present. Let me know which direction you prefer, and I'll make the change. Thanks 🙇🏻

Reply:
>I agree with you that it is a bit annoying. I couldn't immediately think of a way to handle optionally loading the config from the file without changing the contract of loadOcmConfig to only fill in unset values.
Do we need loadOcmConfig to only fill unset values? The way `getOcmConfiguration` currently works, I don't think we do?

>I also now see that loadOcmConfig does not throw an error if the config file is not present.
That's a bit nasty, but also seems intended. It's no big deal here. If the configuration couldn't be loaded, it's an empty configuration, and we will run into an error when trying to establish the connection. I wouldn't worry about it in this PR.

I would prefer leaving it as is in this specific PR, and revert the above block of code to only overwrite the already initiated `Config` object. 

Basically, what we're currently doing is (correct me if I'm wrong here):

```
var config
var tmpConfig

if should_load_ocm_config_file
   tmpConfig = loadOcmConfigFile()

// Check which one has a value with priority and load it into our final config object
// If other values are set by `loadOcmConfigFile`, we lose them.
if OCM_ENV_1 != "" {
    config.ocm_1 = OCM_ENV_1
else {
   config.ocm_1 = tmpConfig.ocm_1
```
vs before
```
var config

if should_load_ocm_config_file
   // Pre-load into the config object
   config = loadOcmConfigFile()

// Overwrite the same config object, in case the env has a value
if OCM_ENV_1 != "" {
    config.ocm_1 = OCM_ENV_1
```

Reply:
Ok, I see you. One minute.

Reply:
Updated; pushed; tests still pass.